listenbrainz/webserver/static/js/src/follow/FollowerFollowingModal.tsx

```diff
@@ -65,24 +73,46 @@ export default class FollowerFollowingModal extends React.Component<
             </Pill>
           </div>
         </div>
-        <div className="follower-following-list">
-          {activeModeList.map((listEntry: string) => {
-            const formattedAsUser: ListenBrainzUser = {
-              name: listEntry,
-            };
-            return (
-              <UserListModalEntry
-                mode="follow-following"
-                key={listEntry}
-                user={formattedAsUser}
-                loggedInUserFollowsUser={loggedInUserFollowsUser(
-                  formattedAsUser
-                )}
-                updateFollowingList={updateFollowingList}
-              />
-            );
-          })}
-        </div>
+
+        {activeModeList.length === 0 ? (
+          <>
+            <hr style={{ margin: "0px 2em", borderTop: "1px solid #eee" }} />
+            {activeMode === "follower" ? (
+              <div className="follower-following-empty text-center text-muted">
+                {user.name === currentUser?.name
+                  ? "You don't"
+                  : `${user.name} doesn't`}{" "}
+                have any followers.
+              </div>
+            ) : (
+              <div className="follower-following-empty text-center text-muted">
+                {user.name === currentUser?.name
+                  ? "You aren't"
+                  : `${user.name} isn't`}{" "}
+                following anyone.
+              </div>
+            )}
+          </>
+        ) : (
```

Comment:
The nested ternary operations get pretty hard to read, I think it would be better to have something like this for legibility:
```jsx
if(activeModeList.length === 0){
  if(activeMode === "follower"){
  // return "no followers" markup
  }

  // return "not following" markup
}
// return regular markup
return <div className="follower-following-list">…
```

Reply:
I agree, nested ternary operations are confusing to read. I was trying to use if-else statements to get it done because it was hurting the readability. But it turns out you can't use normal if-else statements in JSX. It is explained in the following links why is that the case:
1. https://react-cn.github.io/react/tips/if-else-in-JSX.html
2. https://reactjs.org/docs/conditional-rendering.html
3. https://www.digitalocean.com/community/tutorials/7-ways-to-implement-conditional-rendering-in-react-applications

I can try separating it out as a new component though. I'll give you an update on this.

Reply:
Ah, there is a subtlety here: you cannot use if-else statements *inside a JSX expression* meaning in a JSX block (the part that starts with an opening html bracket `<` ) , but you can definitely use them outside of it in the same file.
Any react component after all is just a standard javascript function…

From your first link (emphasis mine):
> If a ternary expression isn't robust enough, you can use if statements *outside of your JSX* to determine which components should be used